pkg/master/master.go

```diff
@@ -87,22 +88,25 @@ const (
 
 // Config is a structure used to configure a Master.
 type Config struct {
-	DatabaseStorage tools.StorageInterface
-	EventTTL        time.Duration
-	MinionRegexp    string
-	KubeletClient   client.KubeletClient
+	DatabaseStorage    tools.StorageInterface
+	ExpDatabaseStorage tools.StorageInterface
+	EventTTL           time.Duration
+	MinionRegexp       string
+	KubeletClient      client.KubeletClient
 	// allow downstream consumers to disable the core controller loops
 	EnableCoreControllers bool
 	EnableLogsSupport     bool
 	EnableUISupport       bool
 	// allow downstream consumers to disable swagger
 	EnableSwaggerSupport bool
-	// allow v1 to be conditionally disabled
-	DisableV1 bool
+	// allow api versions to be conditionally disabled
+	DisableV1  bool
+	DisableExp bool
```

Comment:
I'm not sure I see a reason to. It's unnecessarily hard to enable otherwise and doesn't interfere with normal behavior. I think we could just disable support in kubelet by default.